### PR TITLE
Added basic types: string, IPv4 address, IPv6 address.

### LINKED ISSUE #1183: Basic types
_From Genart review: https://mailarchive.ietf.org/arch/msg/taps/L7_Aopnvp-xn9aEQ_zI9kgxts68/_

IPv4 and IPv6 addresses should be also listed among the basic types?
Also, strings (e.g., used for services, interfaces, hostnames).

### PR DESCRIPTION
**Pull request by mwelzl**
Closes #1183

### CODE REVIEW

**Review by philsbln — CHANGES_REQUESTED** [review]
I have three asks for this PR:
- Please add String/IPv6Address/IPv4Address a little further up, as we sorted simple types first, collection/composed types later
- Do we need to restrict ourselves to alphanumeric strings or shouldn't we ask for the default string type of the implementation, preferably unicode-compatible
- Separate types for IPv4 and IPv6 or better have a type that catches both?

**Comment by mwelzl** [comment]
Done; better now?  @philsbln please check.

File: draft-ietf-taps-interface.md [hunk 1]
```diff
@@ -212,6 +212,9 @@ We also make use of the following basic types:
 - Boolean: Instances take the value `true` or `false`.
 - Integer: Instances take positive or negative integer values.
 - Numeric: Instances take positive or negative real number values.
+- String: Instances are represented in a way that is common to the programming
+  language, e.g. unicode.
```
Comment (tfpauly) [inline draft-ietf-taps-interface.md:216]:
```suggestion
  language, e.g. UTF-8.
```

maybe this?

**Review by tfpauly — APPROVED** [review]

**Review by philsbln — APPROVED** [review]
Yes, perfect. Thank you very much!

Now, we need to make Specifying Endpoint consistent to that…  made an Issue #1210 for that and will file a PR too

**Pull request merged**